app/uk/gov/hmrc/customs/inventorylinking/export/controllers/actionbuilders/AuthAction.scala

```diff
@@ -77,12 +75,12 @@ class AuthAction @Inject()(
   // pure function that tames exceptions throw by HMRC auth api into an Either
   // this enables calling function to not worry about recover blocks
   // returns a Future of Left(Result) on error or a Right(AuthorisedRequest) on success
-  private def authoriseAsCsp[A](implicit vhr: ValidatedHeadersRequest[A]): Future[Either[Result, Option[BadgeIdentifierEoriPair]]] = {
+  private def authoriseAsCsp[A](implicit asf: ApiSubscriptionFieldsRequest[A]): Future[Either[Result, Option[BadgeIdentifierEoriPair]]] = {
     implicit def hc(implicit rh: RequestHeader): HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(rh.headers)
 
     authorised(Enrolment("write:customs-inventory-linking-exports") and AuthProviders(PrivilegedApplication)) {
       Future.successful(eitherMaybeBadgeIdentifierEoriPair)
-    }.recover {
+    }.recover[Either[Result, Option[BadgeIdentifierEoriPair]]] {
       case NonFatal(_: AuthorisationException) =>
         logger.debug("Not authorised as CSP")
```

Comment:
Some more debug statements like this one scattered thru other code paths could be useful.